crates/e2e/tests/e2e/liquidity.rs

```diff
@@ -207,49 +195,80 @@ async fn zero_ex_liquidity(web3: Web3) {
         .unwrap();
 
     let sell_token_balance_after = token_usdc
-        .balance_of(trader_a.address())
+        .balance_of(trader.address())
         .call()
         .await
         .unwrap();
     let buy_token_balance_after = token_usdt
-        .balance_of(trader_a.address())
+        .balance_of(trader.address())
         .call()
         .await
         .unwrap();
 
     assert!(sell_token_balance_before > sell_token_balance_after);
-    assert!(buy_token_balance_after >= buy_token_balance_before + to_wei_with_exp(500, 6));
+    assert!(buy_token_balance_after >= buy_token_balance_before + amount);
 }
 
 fn orders_query_handler(
     query: &OrdersQuery,
     order_creation: OrderCreation,
     zeroex_maker: TestAccount,
-    zeroex_taker: TestAccount,
     zeroex_addr: H160,
     gpv2_addr: H160,
     chain_id: u64,
+    weth_address: H160,
 ) -> Result<Vec<OrderRecord>, ZeroExResponseError> {
     if query.sender == Some(gpv2_addr) {
         let typed_order = Eip712TypedZeroExOrder {
-            maker_token: order_creation.sell_token,
+            maker_token: order_creation.buy_token,
+            taker_token: order_creation.sell_token,
+            // fully covers execution costs
+            maker_amount: order_creation.buy_amount.as_u128() * 3,
+            taker_amount: order_creation.sell_amount.as_u128() * 2,
+            taker_token_fee_amount: 0,
+            maker: zeroex_maker.address(),
+            taker: gpv2_addr,
+            sender: gpv2_addr,
+            fee_recipient: zeroex_addr,
+            pool: H256::default(),
+            expiry: NaiveDateTime::MAX.timestamp() as u64,
+            salt: U256::from(Utc::now().timestamp()),
+        };
+        let usdt_weth_order = Eip712TypedZeroExOrder {
+            maker_token: weth_address,
             taker_token: order_creation.buy_token,
-            maker_amount: order_creation.sell_amount.as_u128() * 2,
-            taker_amount: order_creation.buy_amount.as_u128() * 2,
+            // add sufficient liquidity
+            maker_amount: 1_000_000_000_000_000_000u128,
```

Comment:
FYI this amount comes from `--amount-to-estimate-prices-with`. Probably a good idea to add a comment for the purpose of each order. (e.g. main quote, native price sell_token, native price buy_token).